cpp/src/gandiva/encrypt_utils_test.cc

```diff
@@ -0,0 +1,63 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "gandiva/encrypt_utils.h"
+
+#include <gtest/gtest.h>
+
+#include <unordered_set>
+
+#include "gandiva/execution_context.h"
+
+TEST(TestShaEncryptUtils, TestAesEncryptDecrypt) {
+  gandiva::ExecutionContext ctx;
+  auto ctx_ptr = reinterpret_cast<int64_t>(&ctx);
+
+  auto* key = (unsigned char*)"12345abcde";
```

Comment:
Can you test with large key lengths? Please add test cases with 16, 32, 64, 128, 192, and 256 byte keys

Reply:
Tested with 8, 16, 32, 64, 128, 192 and 256 byte keys.